File: Network Monitor/Notifier.m
//
//  Notifier.m
//  Network Monitor
//
//  Created by HCorbucc on 10/14/14.
//  Copyright (c) 2014 Hugo Corbucci. All rights reserved.
//

#import "Notifier.h"

@implementation Notifier

- (void)announce:(id)message {
    NSUserNotification *notification = [[NSUserNotification alloc] init];
    notification.title = @"Network Notifier";
    notification.informativeText = message;
    
    [[NSUserNotificationCenter defaultUserNotificationCenter] deliverNotification: notification];
}

@end

File: Network Monitor/Notifier.swift
//
//  Notifier.swift
//  Network Monitor
//
//  Created by HCorbucc on 10/20/14.
//  Copyright (c) 2014 Hugo Corbucci. All rights reserved.
//

import Foundation

class Notifier: AlertCenter {
    let center : NSUserNotificationCenter
    init() {
        center = NSUserNotificationCenter.defaultUserNotificationCenter()
    }
    func announce(message: NSString) {
        let notification = NSUserNotification()
        notification.title = NSLocalizedString("AppName", comment: "The app name for notification center title")
        notification.informativeText = message
        
        center.deliverNotification(notification)
    }
}